samza-kv/src/main/scala/org/apache/samza/storage/kv/KeyValueStorageEngine.scala

```diff
@@ -124,7 +124,7 @@ class KeyValueStorageEngine[K, V](
     val batch = new java.util.ArrayList[Entry[Array[Byte], Array[Byte]]](batchSize)
     var lastBatchFlushed = false
 
-    while(iterator.hasNext) {
+    while(iterator.hasNext && !Thread.currentThread().isInterrupted) {
```

Comment:
Is it feasible to unit test this?

Reply:
Absolutely. I added a test to verify it bails out as soon as it receives an interrupt.